### PR TITLE
FEATURE: Added search endpoint

### PR DESCRIPTION
**Pull request by dmytrovb02**

### CODE REVIEW

File: src/main/java/com/bookstore/repository/book/BookRepository.java [hunk 1]
```diff
@@ -1,10 +1,11 @@
-package com.bookstore.repository;
+package com.bookstore.repository.book;
 
 import com.bookstore.model.Book;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.stereotype.Repository;
 
 @Repository
```
Comment (Sanderfuck) [inline src/main/java/com/bookstore/repository/book/BookRepository.java]: This anno not required

File: src/main/java/com/bookstore/repository/book/BookSpecificationBuilder.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import com.bookstore.repository.SpecificationBuilder;
+import com.bookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+
```
Comment (Sanderfuck) [inline src/main/java/com/bookstore/repository/book/BookSpecificationBuilder.java]:
```suggestion
```

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookstore/repository/book/BookRepository.java (hunk 1 above)
Comment (dmytrovb02) [reply src/main/java/com/bookstore/repository/book/BookRepository.java]: Yes, but I think for better code readability we can keep this annotation. 
But if you insist I can remove it

File: src/main/java/com/bookstore/repository/book/BookRepository.java [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
```
Comment (koval-kn) [inline src/main/java/com/bookstore/repository/book/BookRepository.java]: annotations dont work for interfaces
```suggestion
```

**Review by koval-kn — APPROVED** [review]

**Pull request merged**